openpype/hosts/maya/plugins/publish/validate_gltf_textures_names.py

```diff
@@ -0,0 +1,126 @@
+                               "nodes need to be connected")
+        invalid = self.get_texture_name_invalid(instance)
+        if invalid:
+            raise RuntimeError("Invalid texture name(s) found: "
+                               "{0}".format(invalid))
+
+    def get_texture_name_invalid(self, instance):
+
+        invalid = set()
+        shading_grp = self.shader_selection(instance)
+
+        # get the materials related to the selected assets
+        # get the file textures related to the PBS Shader
+        # validate the names of the textures
+        for material in shading_grp:
+            main_shader = cmds.listConnections(material,
+                                               destination=True,
+                                               type="StingrayPBS")
+            for shader in main_shader:
+                albedo = cmds.listConnections(shader + ".TEX_color_map")[0]
+                dif_path = cmds.getAttr(albedo + ".fileTextureName")
+                dif = dif_path.split(".")[0]
+                # "_D"
+                if not dif.endswith("_D"):
+                    invalid.add(dif_path)
+                orm_packed = cmds.listConnections(shader + ".TEX_ao_mapX")[0]
+                if orm_packed:
+                    # "_ORM"
+                    orm_path = cmds.getAttr(orm_packed + ".fileTextureName")
+                    orm = orm_path.split(".")[0]
+                    if not orm.endswith("_ORM"):
+                        invalid.add(orm_path)
+                nrm = cmds.listConnections(shader + ".TEX_normal_map")[0]
+                if nrm:
+                    nrm_path = cmds.getAttr(nrm + ".fileTextureName")
+                    nrm_map = nrm_path.split(".")[0]
+                    # "_N"
+                    if not nrm_map.endswith("_N"):
+                        invalid.add(nrm_path)
+
+        return list(invalid)
+
+    def get_texture_node_invalid(self, instance):
+        invalid = set()
+        shading_grp = self.shader_selection(instance)
+        for material in shading_grp:
+            main_shader = cmds.listConnections(material,
+                                               destination=True,
+                                               type="StingrayPBS")
+            for shader in main_shader:
+                # diffuse texture file node
+                albedo = cmds.listConnections(shader + ".TEX_color_map")
+                if not albedo:
+                    invalid.add(albedo)
+        return list(invalid)
+
+    def get_texture_shader_invalid(self, instance):
+
+        invalid = set()
+        shading_grp = self.shader_selection(instance)
+        for material in shading_grp:
+            main_shader = cmds.listConnections(material,
+                                               destination=True,
+                                               type="StingrayPBS")
+            if not main_shader:
+                invalid.add(material)
```

Comment:
Also are you sure the logic here is correct? Usually the shadingGroup is on the source side of the shading groups.

Additionally it feels like you're mixing shading groups/engines and materials making the variables confusing.
The shading engine / shader group is usually the shader. (List under Shading Groups in Hypershade)
The material is what connects into it, e.g. the StingrayPBS material. (Listed under Materials in Hypershade) 

```suggestion
        shading_groups = self.shader_selection(instance)
        for shading_group in shading_groups :
            material = cmds.listConnections(shading_group,
                                            source=True,
                                            destination=False,
                                            type="StingrayPBS")
            if not material:
                invalid.add(material)
```

I think you're even looking for materials that are connected to the shading group's `surfaceShader` attribute so it might be good to explicitly check that toon, e.g. by doing this:

```python
material = cmds.listConnections("{}.surfaceShader".format(shading_group),
                                source=True,
                                destination=False,
                                type="StingrayPBS")
```

Reply:
I need to admit I need more coffee for this particular case, will do the adjustment regarding to that